### PR TITLE
add 2.0 upgrade guide

### PR DESCRIPTION
**Pull request by ms-henglu**
The `2.0-upgrade-guide` in the `doc` folder is synced from the one in the `templates`. The two files have same contents.

### CODE REVIEW

File: docs/guides/2.0-upgrade-guide.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+provider "azapi" {
+}
+```
+
+.. or to any 1.x release:
+
+```hcl
+terraform {
+  required_providers {
+    azurerm = {
```
Comment (cwe1ss) [inline docs/guides/2.0-upgrade-guide.md:44]: I think this should be "azapi" instead of "azurerm"

**Review by magodo — APPROVED** [review]

File: docs/guides/2.0-upgrade-guide.md [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+---
+
+
+## What's available in Version 2.0 of the Azapi Provider?
+
+Below is an overview of the changes coming 2.0. Each topic is covered in more detail further down.
+
+* [Dynamic Properties Support](#dynamic-properties-support)
+* [Provider Functions](#provider-functions)
+* [Cusomtized Retry Configuration](#customized-retry-configuration)
```
Comment (stemaMSFT) [inline docs/guides/2.0-upgrade-guide.md:66]:
```suggestion
* [Customized Retriable Configuration](#customized-retry-configuration)
```

File: docs/guides/2.0-upgrade-guide.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    ]
+  }
+}
+
+output "o1" {
+  // it's not necessary to use `jsondecode` function to decode the response
+  value = azapi_resource.automationAccount.output.properties.automationHybridServiceUrl
+}
+```
+
```
Comment (stemaMSFT) [inline docs/guides/2.0-upgrade-guide.md:113]: ``
```suggestion
For more information, refer to [this blogpost](https://techcommunity.microsoft.com/t5/azure-tools-blog/announcing-azapi-dynamic-properties/ba-p/4121855)
```

File: docs/guides/2.0-upgrade-guide.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+- Provider field `default_naming_prefix` and `default_naming_suffix` are removed. 
+
+  How to fix:  
+  Please remove the `default_naming_prefix` and `default_naming_suffix` fields from the provider block. You can specify the naming prefix and suffix in the resource's `name` field instead.
+
+- Provider field `enable_hcl_output_for_data_source` is removed. The `output` field in the data source is always in HCL format.
+
+  How to fix:  
+  Please remove the `enable_hcl_output_for_data_source` field from the provider block. And remove the `jsondecode` function when using the `output` field in the data source.
```
Comment (stemaMSFT) [inline docs/guides/2.0-upgrade-guide.md:377]:
```suggestion
  Please remove the `enable_hcl_output_for_data_source` field from the provider block. Remove any `jsondecode` functions when using the `output` field in the data source.
```

File: docs/guides/2.0-upgrade-guide.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+- The `azapi_resource`'s `removing_special_chars` field is removed. Please specify the `name` field and remove the special characters in the `name` field instead.
+
+  How to fix:  
+  Please remove the `removing_special_chars` field from the `azapi_resource` block. And remove the special characters in the `name` field.
```
Comment (stemaMSFT) [inline docs/guides/2.0-upgrade-guide.md:382]:
```suggestion
  Please remove the `removing_special_chars` field from the `azapi_resource` block and any special characters in the `name` field.
```

**Review by stemaMSFT — COMMENTED** [review]
LGTM, just a few minor changes!

**Comment by ms-henglu** [comment]
Thanks @cwe1ss and @stemaMSFT for the code review! I've updated this PR as suggested.

**Review by stemaMSFT — APPROVED** [review]

**Pull request merged**